### DISCUSSION TITLE
How to reuse same process within a workflow?

### DISCUSSION THREAD

**Discussion by jfxer** [nextflow-io/nextflow#1949] [Q&A] [answered]
## Bug report 
workflow {
     HELLO()
     HELLO()
}
### Expected behavior and actual behavior
expected behavior: Run process HELLO twice
actual result : Process 'HELLO' has been already used -- If you need to reuse the same component include it with a different name or include in a different workflow context

* Nextflow version: 20.10.0 
* Java version: 8
* Operating system: [Ubuntu 18.04]
* Bash version: 5.0.17(1)-release (x86_64-pc-linux-gnu)

### Additional context

DSL 2 is enabled.

is I have any way to run the processes on loop?
something like:
    workflow {
     for (int i=0; I< 5; i++){
           HELLO()
     }
}

**Comment by abhi18av** [comment, marked as answer]
Hi @jfxer ,

Actually, this aspect of DSL2 seems to be by design 😉 

Perhaps, [module aliasing](https://www.nextflow.io/docs/latest/dsl2.html#module-aliases) section of the docs might be of help in your use-case.

Regarding the second part of the question, which is independent of the first one. This seems contrary to the `DataFlow` paradigm adapted by NF, which emphasizes parallel processing  (via channels) rather than serial processing (via loops).

**Reply by jfxer** [reply]
@abhi18av Thanks for the answer. Is any way to restart some process maybe using channels or any other way?(From another process or workflow). Alias is not fit in reason of the big amount of restarts.

**Reply by abhi18av** [reply]
@jfxer , as this is a question of design of the pipeline, I suggest that it's best to discuss this further in the context of a working script or a repository - what do you think?

Could you please initiate the repo/share functional snippet of what you'd like to achieve overall.

**Reply by jfxer** [reply]
@abhi18av Hi. There  https://github.com/jfxer/nextflow_sample is a simplified version of the steps of QC which works with files and should be run until all QC steps didn't remove any line from a file(On some iteration).
